### PR TITLE
Add instance_norm op and test.

### PR DESCRIPTION
**Pull request by asenina**
### Changes
The  torch.nn.InstanceNorm3d layer is wrapping by NNCF. 

### Reason for changes
This is necessary for correct pruning of models with conv3d and InstanceNorm3d

### Related tickets
#77188

### Tests

 tests/torch/pruning/test_model_pruning_analysis.py

### CODE REVIEW

File: nncf/torch/graph/operator_metatypes.py [hunk 1]
```diff
@@ -758,6 +758,12 @@ class PTSumMetatype(PTOperatorMetatype):
     }
     hw_config_names = [HWConfigOpName.REDUCESUM]
 
+@PT_OPERATOR_METATYPES.register()
+class PTInstanceNormMetatype(PTOperatorMetatype):
```
Comment (kshpv) [inline nncf/torch/graph/operator_metatypes.py:762]: Need to extend a fusable patterns with new NormLayer

File: nncf/torch/layers.py [hunk 2]
```diff
@@ -156,6 +156,19 @@ def from_module(module):
         return nncf_bn
 
 
+class NNCFInstanceNorm(_NNCFModuleMixin, nn.InstanceNorm3d):
```
Comment (kshpv) [inline nncf/torch/layers.py:159]: Why only nn.InstanceNorm3d?

**Review by kshpv — CHANGES_REQUESTED** [review]

File: nncf/torch/layers.py (hunk 2 above)
Comment (asenina) [reply nncf/torch/layers.py:159]: The target of this PR was to support a specific model(with only `nn.InstanceNorm3d`) from the ticket.
But i see no problem for adding `nn.InstanceNorm1d`, `nn.InstanceNorm2d`. It is also strange that we only wrap `nn.BatchNorm2d`)

Comment (vshampor) [reply nncf/torch/layers.py:159]: At the very least you should call the wrapper NNCFInstanceNorm**3d**.

**Review by vshampor — CHANGES_REQUESTED** [review]

File: tests/torch/pruning/helpers.py [hunk 3]
```diff
@@ -669,6 +670,27 @@ def forward(self, x):
...
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.conv0(x)
+        x1 = self.in0(x)
+        x1 = F.leaky_relu(x1)
+        x1 = self.conv1(x1)
+        x1 = self.in1(x1)
+        x1 = F.leaky_relu(x1)
+        x1 = self.conv2(x1)
+
+        return x + x1
```
Comment (daniil-lyakhov) [inline tests/torch/pruning/helpers.py:691]: Please consider to add conv layer to the end of model to enable first pruning group to cover all cases of the unet block pruning

File: nncf/torch/pruning/base_algo.py [hunk 4]
```diff
@@ -138,7 +138,7 @@ def _prune_weights(self, target_model: NNCFNetwork):
                                  PTNNCFPruningTensorProcessor).mask_propagation()
 
         # Adding binary masks also for Batch/Group Norms to allow applying masks after propagation
-        types_to_apply_mask = ['group_norm']
+        types_to_apply_mask = ['group_norm', 'instance_norm']
```
Comment (daniil-lyakhov) [inline nncf/torch/pruning/base_algo.py:141]: Please check masks are correctly applying to instance_norm weights during training

File: tests/torch/pruning/helpers.py (hunk 3 above)
Comment (asenina) [reply tests/torch/pruning/helpers.py]: Done

**Comment by asenina** [comment]
Jenkins please retry a build

File: nncf/torch/graph/pattern_operations.py [hunk 5]
```diff
@@ -34,6 +34,14 @@
 GROUP_NORMALIZATION_OPERATIONS = {'type': ['group_norm'],
                                   'label': 'GROUP_NORMALIZATION'}
 
+INSTANCE_NORMALIZATION_OPERATIONS = {'type': ['instance_norm',
+                                              'instance_norm1d',
+                                              'instance_norm2d',
+                                              'instance_norm3d'
+                                           ],
+                                  'label': 'INSTANCE_NORMALIZATION'
+                                  }
```
Comment (dupeljan) [inline nncf/torch/graph/pattern_operations.py:43]: Please format this

File: nncf/torch/pruning/base_algo.py (hunk 4 above)
Comment (dupeljan) [reply nncf/torch/pruning/base_algo.py:142]: Resolved

**Review by dupeljan — APPROVED** [review]

**Review by dupeljan — APPROVED** [review]
Sorry, this is Daniil-Lyakhov, I can't access my account now due to phone verification problems

**Review by daniil-lyakhov — APPROVED** [review]

**Comment by asenina** [comment]
Jenkins please retry a build

**Review by vshampor — APPROVED** [review]

**Review by kshpv — APPROVED** [review]

File: nncf/torch/graph/pattern_operations.py [hunk 6]
```diff
@@ -45,6 +45,11 @@
     GraphPattern.LABEL_ATTR: "LAYER_NORMALIZATION",
 }
 
+INSTANCE_NORMALIZATION_OPERATIONS = {
+    "type": ["instance_norm", "instance_norm1d", "instance_norm2d", "instance_norm3d"],
+    "label": "INSTANCE_NORMALIZATION",
```
Comment (daniil-lyakhov) [inline nncf/torch/graph/pattern_operations.py:50]:
```suggestion
   GraphPattern.LABEL_ATTR: "INSTANCE_NORMALIZATION",
```

**Review by daniil-lyakhov — APPROVED** [review]

File: nncf/torch/graph/pattern_operations.py (hunk 6 above)
Comment (vshampor) [reply nncf/torch/graph/pattern_operations.py]: Done

**Comment by vshampor** [comment]
@daniil-lyakhov the pruning tests are failing, please see what's wrong and advise, or directly push additional fixing commits into this PR.

**Comment by daniil-lyakhov** [comment]
@vshampor, I tried to find out reasons, but after an our I think one need to do a deep dive into pruning algo again to solve this. I have no clue why references have changed after LayerNorms presentation. I suggest to close this PR, otherwise, I feel like I need some approve to spend a day to understand the root cause
